tests/unit/Lookup/EntityRetrievingClosestReferencedEntityIdLookupTest.php

```diff
@@ -0,0 +1,476 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Tests\Lookup;
+
+use DataValues\StringValue;
+use PHPUnit_Framework_TestCase;
+use Wikibase\DataModel\Entity\EntityId;
+use Wikibase\DataModel\Entity\EntityIdValue;
+use Wikibase\DataModel\Entity\Item;
+use Wikibase\DataModel\Entity\ItemId;
+use Wikibase\DataModel\Entity\PropertyId;
+use Wikibase\DataModel\Services\Entity\EntityPrefetcher;
+use Wikibase\DataModel\Services\Entity\NullEntityPrefetcher;
+use Wikibase\DataModel\Services\Lookup\EntityLookup;
+use Wikibase\DataModel\Services\Lookup\EntityLookupException;
+use Wikibase\DataModel\Services\Lookup\EntityRetrievingClosestReferencedEntityIdLookup;
+use Wikibase\DataModel\Services\Lookup\InMemoryEntityLookup;
+use Wikibase\DataModel\Services\Lookup\MaxReferenceDepthExhaustedException;
+use Wikibase\DataModel\Services\Lookup\MaxReferencedEntityVisitsExhaustedException;
+use Wikibase\DataModel\Services\Lookup\ReferencedEntityIdLookupException;
+use Wikibase\DataModel\Services\Lookup\RestrictedEntityLookup;
+use Wikibase\DataModel\Snak\PropertyNoValueSnak;
+use Wikibase\DataModel\Snak\PropertyValueSnak;
+use Wikibase\DataModel\Statement\Statement;
+use Wikibase\DataModel\Statement\StatementList;
+
+/**
+ * @covers Wikibase\DataModel\Services\Lookup\EntityRetrievingClosestReferencedEntityIdLookup
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ *
+ * @license GPL-2.0-or-later
+ * @author Marius Hoch
+ */
+class EntityRetrievingClosestReferencedEntityIdLookupTest extends PHPUnit_Framework_TestCase {
+
+	/**
+	 * @param EntityLookup $entityLookup
+	 * @param int $expectedNumberOfGetEntityCalls
```

Comment:
Why have this argument an an if-else when you can just have two simple methods?

Reply:
True, can even be one simpler method… changed to just use phpunit mocks all the time and not sometimes rely on `RestrictedEntityLookup`.